src/Illuminate/Queue/Console/WorkCommand.php

```diff
@@ -182,14 +182,11 @@ protected function listenForEvents()
      */
     protected function writeOutput(Job $job, $status)
     {
-        switch ($status) {
-            case 'starting':
-                return $this->writeStatus($job, 'Processing', 'comment');
-            case 'success':
-                return $this->writeStatus($job, 'Processed', 'info');
-            case 'failed':
-                return $this->writeStatus($job, 'Failed', 'error');
-        }
+        match ($status) {
```

Comment:
you need to `return` the result of the `match` to mimic the previous functionality

Reply:
the writeStatus method has void type and does not return anything already

Reply:
correct, but you need to match the behavior of the existing `writeOutput` method, which `return`s the result of the `writeStatus` method. `writeStatus()` returning `void` is irrelevant.

Reply:
nothing to change here.

Reply:
strongly disagree. if you didn't already have the docblock, or where able to look into the `writeStatus` method, you would have no clue that it currently returns `void`. Therefore, if you want to match functionality in this PR targeting the current major version, you **need** to `return` the result of the call, just like we did prior.

If you want to change functionality, which is what dropping `return` would be, target 10.x.

Reply:
I strongly disagree with your strong disagreement :smile:

The `writeStatus` method is clearly documented as `void`. As Laravel does not have (especially scalar) typehints in most of its codebase the docblock is what defined the API contract of the method. If someone overrode the `writeStatus` method and made it actually return anything is in breach of that contract. So it should not be in Laravel backwards compatibility policy to care about "breaking" use cases that were never officially supported/allowed to begin with.

The only reason why there's a return in that switch statement is to not have to add an additional `break;` line, not to actually return some value.